README.md

```diff
@@ -1 +1,57 @@
-# riscv
+# RV32G Processor model
+
+This is the functional model of RISC-V instruction set architecture. It
+contains all the instructions necessary for running mibench programs using the 
+base interger model. It has following types of instruction encoding:
+
+  - Integer Base instructions
+  - Single-Precision floating point instructions 
+  - Double-Precision floating point instructions
+
+## Installation
+
+For installing the processor model follow the below steps:
+
+  - Step 1:. Download the RISC-V processor model in the processors folder(made 
+             during the installation of ArchC) 
+  - Step 2:. Perform the following commands on terminal: 
+`````````
+acsim riscv.ac -abi     (Create the simulator)     
+make                    (Compile it) 
+`````````
+
+## How to create executables
+
+Follow the below steps:
+
+  - Step 1:. Go to the tests->libac_sysc folder and run make command
+via terminal.
+  - Step 2:. Copy the folder of the program you wish to test from 
+test->acstone-prgrams or test->automotive-IMA or test->acstone-FP to the test 
+folder. Enter that folder and run make via terminal.
+
+## Running the executable
+
+For running the test program on the model perform the following on the terminal:
+````````` 
+./riscv.x –load=<path/to/the/executable>.run  
+`````````
+## Bugs in the model
+Following bugs are there in the model:
+
+  - FLD and FSD always return 0.00
+
+## Future Work
+
+Following things are left and need to be developed:
+
+   - Developing syscalls such that input arguments can be given by the user 
+
+## References
+
+For developing my model I used the following as a reference:
+   - Mips processor model
+
+## Code Merged
+
+Dario's framework developed for creating the executables has been used.
```

Comment:
I though you have used more things from Dario's model. If you have, put here "Based on Dario's RISC-V model", put his full name and, if available, github user.